demos/common/processing/config_factory.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+// Copyright (C) 2018-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "config_factory.h"
+
+#include <set>
+
+#include <samples/args_helper.hpp>
+#include <samples/common.hpp>
+#include <cldnn/cldnn_config.hpp>
+
+DECLARE_string(d);
+DECLARE_bool(pc);
+DECLARE_string(c);
+DECLARE_string(l);
+DECLARE_uint32(nireq);
+DECLARE_uint32(nthreads);
+DECLARE_string(nstreams);
```

Comment:
Pass flags values as arguments instead.

Reply:
The idea of configs factory is that it should construct config from command line parameters directly. Passing parameters as arguments is an extra effort here.

Reply:
Well, to me an extra effort is declaring these flags here. The reason I'm asking about it is to make function interface represent what it depends on.

Reply:
As I wrote, one of main purposes of this class is to use command line data directly. We'll replace these macros after implementing better class-based approach for command line parsing.

Reply:
So you should pass command line data directly as arguments.

Reply:
No. It works exactly as designed.

Reply:
Well, the design is questionable:
> make function interface represent what it depends on.

Reply:
gFlags do not allow to pass data as single object. Passing them as multiple variable is even worse solution leading to code increase and making it more confusing. So choosing between these 2 solutions, I choose current one. Anyway, this part of code will be changed as soon as we introduce better command line parser.

Reply:
done